File: app/src/main/java/yangfentuozi/runner/service/data/CommandInfo.java
package yangfentuozi.runner.service.data;

import android.os.Parcel;
import android.os.Parcelable;

public class CommandInfo implements Parcelable {

    public String name;
    public String command;
    public boolean keepAlive;
    public boolean reducePerm;
    public String targetPerm;

    public CommandInfo() {
    }

    public CommandInfo(Parcel source) {
        super();
        name = source.readString();
        command = source.readString();
        keepAlive = source.readInt() == 1;
        reducePerm = source.readInt() == 1;
        targetPerm = source.readString();
    }

    public int describeContents() {
        return 0;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(command);
        dest.writeInt(keepAlive ? 1 : 0);
        dest.writeInt(reducePerm ? 1 : 0);
        dest.writeString(targetPerm);
    }

    public static final Parcelable.Creator<CommandInfo> CREATOR = new Parcelable.Creator<>() {
        public CommandInfo createFromParcel(Parcel source) {
            return new CommandInfo(source);
        }

        public CommandInfo[] newArray(int size) {
            return new CommandInfo[size];
        }
    };
}

File: app/src/main/java/yangfentuozi/runner/shared/data/CommandInfo.kt
package yangfentuozi.runner.shared.data

import android.os.BaseBundle
import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable

open class CommandInfo : Parcelable {
    var name: String? = null
    var command: String? = null
    var keepAlive: Boolean = false

    constructor()

    constructor(source: Parcel) : super() {
        name = source.readString()
        command = source.readString()
        keepAlive = source.readInt() == 1
    }

    constructor(bundle: BaseBundle) : super() {
        name = bundle.getString("name")
        command = bundle.getString("command")
        keepAlive = bundle.getBoolean("keepAlive", false)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(command)
        dest.writeInt(if (keepAlive) 1 else 0)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<CommandInfo?> = object : Parcelable.Creator<CommandInfo?> {
            override fun createFromParcel(source: Parcel): CommandInfo {
                return CommandInfo(source)
            }

            override fun newArray(size: Int): Array<CommandInfo?> {
                return arrayOfNulls<CommandInfo>(size)
            }
        }
    }

    fun toBundle(): Bundle {
        return Bundle().apply {
            putString("name", name)
            putString("command", command)
            putBoolean("keepAlive", keepAlive)
        }
    }
}